query-service-impl/src/test/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverterTest.java

```diff
@@ -0,0 +1,218 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import static java.util.Objects.requireNonNull;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createColumnExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createFunctionExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createInFilter;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createStringLiteralValueExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createTimeColumnGroupByExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createTimeFilter;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.util.LinkedHashSet;
+import java.util.List;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.api.QueryRequest.Builder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class QueryRequestToPromqlConverterTest {
+
+  private static final String TENANT_COLUMN_NAME = "tenant_id";
+
+  private static final String TEST_REQUEST_HANDLER_CONFIG_FILE = "prometheus_request_handler.conf";
+
+  @Test
+  public void testInstantQueryWithGroupByWithMultipleAggregates() {
+    QueryRequest orderByQueryRequest = buildMultipleGroupByMultipleAggQuery();
+    Builder builder = QueryRequest.newBuilder(orderByQueryRequest);
+    builder.setLimit(20);
+    PrometheusViewDefinition prometheusViewDefinition = getDefaultPrometheusViewDefinition();
+
+    QueryRequest queryRequest = builder.build();
+
+    ExecutionContext executionContext = new ExecutionContext("__default", queryRequest);
+    executionContext.setTimeFilterColumn("SERVICE.startTime");
+    PromqlQuery promqlQuery =
+        new QueryRequestToPromqlConverter(prometheusViewDefinition)
+            .toPromql(
+                executionContext, builder.build(), createSelectionsFromQueryRequest(queryRequest));
```

Comment:
Can we extract this out in a function ?

Reply:
This will be added in another pr.

Reply:
Since assertions in all the tests have a similar pattern, we can create an `assertPromqlQuery` method ?

Reply:
I misread the comment my bad, for this one I purposefully kept the assertion in the same place, ow it gets difficult to read the test.